packages/ra-core/src/CoreAdminRouter.tsx

```diff
@@ -122,20 +159,22 @@ export class CoreAdminRouter extends Component {
             return <Route path="/" key="loading" component={loading} />;
         }
 
-        let childrenToRender =
+        const childrenToRender =
             typeof children === 'function' ? this.state.children : children;
 
         return (
             <div>
                 {// Render every resources children outside the React Router Switch
                 // as we need all of them and not just the one rendered
-                Children.map(childrenToRender, child =>
-                    cloneElement(child, {
-                        key: child.props.name,
-                        // The context prop instructs the Resource component to not render anything
-                        // but simply to register itself as a known resource
-                        context: 'registration',
-                    })
+                Children.map(
+                    childrenToRender,
+                    (child: React.ReactElement<any>) =>
```

Comment:
We need a type here. We're passing a context props, we must check that the child accepts it

Reply:
You're aware that won't be used anywhere outside this component ?